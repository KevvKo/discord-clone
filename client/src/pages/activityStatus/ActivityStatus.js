import React from 'react';
// Components
import SettingsPanel from '../../components/settingsPanel/SettingsPanel';
// Hooks
import { useTranslation } from 'react-i18next';

function ActivityStatus(){

    const [ t ] = useTranslation('common');

    return(
        <SettingsPanel>
            <h3>{t('settings.main.activityStatus.title')}</h3>
        </SettingsPanel>
    );
}

export default ActivityStatus;