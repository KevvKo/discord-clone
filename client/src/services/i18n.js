import i18next from "i18next";
import { initReactI18next } from "react-i18next";
import commonDe from '../configs/translations/de/common.json'
import commonEn from '../configs/translations/en/common.json'

const resources = {
  en: {
      common: commonEn
  },
  de: {
      common: commonDe
  }
}

i18next
  .use(initReactI18next)
  .init({
    resources,
    lng: "en",
    interpolation: {
      escapeValue: false,
    },
  });

export default i18next;