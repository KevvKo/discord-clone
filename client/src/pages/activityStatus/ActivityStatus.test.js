import React from 'react';
import ActivityStatus from './ActivityStatus';
import renderer from 'react-test-renderer';
import '../../scripts/i18n';

describe('ActivityStatus', () => {
    let component;

    beforeEach(() => {
        component = renderer
        .create( <ActivityStatus />)
    })
    it('renders the component correctly', () => {

        component.toJSON();
        expect(component).toMatchSnapshot();
    });
    it('renders ActivityStatus title correctly', () => {
      expect(component.root.findByType('h3').children[0]).toBe('Activity Status');
    });
});
