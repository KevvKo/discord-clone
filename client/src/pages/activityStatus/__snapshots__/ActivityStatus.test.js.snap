// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityStatus renders the component correctly 1`] = `
<div
  className="settings-panel flex-column mt-5 px-4 overflow-y-auto"
>
  <h3>
    Activity Status
  </h3>
</div>
`;
